lab_5_scrapper/scrapper.py

```diff
@@ -90,25 +200,40 @@ def __init__(self, config: Config) -> None:
         """
         Initializes an instance of the Crawler class
         """
-        pass
+        self._config = config
+        self._seed_urls = config.get_seed_urls()
+        self.urls = []
 
     def _extract_url(self, article_bs: BeautifulSoup) -> str:
         """
         Finds and retrieves URL from HTML
         """
-        pass
+        url = article_bs.get('href')
+        if url and url.startswith('https://kazanfirst.ru/news/') and url.count('/') == 5:
+            return url
 
     def find_articles(self) -> None:
         """
         Finds articles
         """
-        pass
+        for seed_url in self._seed_urls:
+            response = make_request(seed_url, self._config)
+            if response.status_code != 200:
+                continue
+            article_bs = BeautifulSoup(response.text, 'lxml')
+            pages = article_bs.find_all(
+                'a', {'class': 'post-block-item  post-item column-list__item js-column-item'})
+            for page in pages:
+                url = self._extract_url(page)
+                self.urls.append(url)
```

Comment:
can url be not a valid url? None? empty string? why do not you check

Reply:
done, checked